roles/matrix/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+- name: Create virtualenv
+  command: virtualenv -p python2.7 ~/.synapse
+  args:
+    chdir: /home/{{ matrix_username }}
+
+- name: PIP to upgrade setup tools
+  pip: name=setuptools virtualenv=/home/{{ matrix_username }}/.synapse extra_args='--upgrade'
+
+- name: PIP to install needed dependencies into virtualenv
+  action: pip name={{ item }} virtualenv=/home/{{ matrix_username }}/.synapse
+  with_items:
+   - cryptography
+   - pyasn1_modules
+   - characteristic
+   - simplejson
+   - canonicaljson
+
+- name: PIP to install synapase from Git Repo into virtualenv
+  pip: name={{ git_repo }} virtualenv=/home/{{ matrix_username }}/.synapse
+
+- name: Generate homeserver.yaml config file
```

Comment:
Wouldn't it be cleaner to add `homeserver.yaml` to Ansible? Or is this done this way because you expect a lot of changes to the config at this stage of development of Matrix?